src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -2745,6 +2745,11 @@ def set_nsg_flow_logging(cmd, client, watcher_rg, watcher_name, nsg, storage_acc
     if retention is not None:
         RetentionPolicyParameters = cmd.get_models('RetentionPolicyParameters')
         config.retention_policy = RetentionPolicyParameters(days=retention, enabled=int(retention) > 0)
+    if cmd.supported_api_version(min_api='2018-10-01') and (log_format or log_version):
```

Comment:
According to the [swagger](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/networkWatcher.json#L2621), it seems that the service-end default for version is  0. However, type has no default. Should we permit the user to specify a version but no log_format? If so, what happens?

Reply:
The only acceptable format is JSON so whether they specify it or not, it's that. The only acceptable versions, ironically, are 1 and 2... so the default of 0 ends up just being mapped to 1 by the service.